# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/DocAccessibleChild.h
# Commit: 0f241c48730c
# Full Hash: 0f241c48730c6e4cce24bc9970408087045ca2ab
# Author: James Teh <jteh@mozilla.com>
# Date: 2020-03-14 10:28:42
# Regressor Bug: 1614871
# File Overlap Count: 1
# Description:
#   Bug 1614871 part 1: Make non-Windows ProxyAccessible::ChildAtPoint cross into in-process iframes. r=eeejay
#   
#   Accessible::ChildAtPoint can return an Accessible in a descendant document.
#   Thus, we must return the result PDocAccessible via IPC, not just the id.
#   Previously, we only returned the id, but we'd fail when we tried to look it up if it belonged to a descendant document.
# ==============================================================================

diff -r 51fa8835f5d8 -r 0f241c48730c accessible/ipc/other/DocAccessibleChild.h
--- a/accessible/ipc/other/DocAccessibleChild.h	Fri Mar 13 01:42:05 2020 +0000
+++ b/accessible/ipc/other/DocAccessibleChild.h	Fri Mar 13 06:14:30 2020 +0000
@@ -461,8 +461,8 @@
 
   virtual mozilla::ipc::IPCResult RecvAccessibleAtPoint(
       const uint64_t& aID, const int32_t& aX, const int32_t& aY,
-      const bool& aNeedsScreenCoords, const uint32_t& aWhich, uint64_t* aResult,
-      bool* aOk) override;
+      const bool& aNeedsScreenCoords, const uint32_t& aWhich,
+      PDocAccessibleChild** aResultDoc, uint64_t* aResultID) override;
 
   virtual mozilla::ipc::IPCResult RecvExtents(const uint64_t& aID,
                                               const bool& aNeedsScreenCoords,